src/reference_implementations/prompting_vector_llms/transferring_gradient_optimized_prompts/transferring_optimized_prompts.ipynb

```diff
@@ -91,50 +198,81 @@
        " 'label': 1}"
       ]
      },
-     "execution_count": 3,
+     "execution_count": 5,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
     "dataset = load_dataset(\"sst2\", split=\"validation\")\n",
-    "dataloader = DataLoader(dataset, batch_size=10, shuffle=True)\n",
+    "dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=True)\n",
     "dataset[0]"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In order to facilitate answer extraction, we need to know the tokenization ids for each of the labels we expect the model to respond with. In this case, we optimized the prompts for T5 to produce either \"negative\" or \"positive.\" So we use the tokenizers to extract the IDs associated with these labels."
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "[2, 40025, 877, 5, 5702, 9, 5, 220, 3645, 4, 38, 657, 42, 1569, 328, 2430]\n",
-      "[2, 40025, 877, 5, 5702, 9, 5, 220, 3645, 4, 38, 657, 42, 1569, 328, 1313]\n",
+      "[1, 3251, 403, 278, 19688, 310, 278, 2446, 10541, 29889, 306, 5360, 445, 14064, 29991, 8178]\n",
+      "[1, 3251, 403, 278, 19688, 310, 278, 2446, 10541, 29889, 306, 5360, 445, 14064, 29991, 6374]\n",
       "[6939, 2206, 8, 6493, 13, 8, 416, 7142, 5, 27, 333, 48, 1974, 55, 2841, 1]\n",
       "[6939, 2206, 8, 6493, 13, 8, 416, 7142, 5, 27, 333, 48, 1974, 55, 1465, 1]\n"
      ]
     }
    ],
    "source": [
-    "# Need to grab the token id associated with the label words for both opt and t5\n",
-    "\n",
     "dummy_sentence = \"I love this movie!\"\n",
-    "print(opt_tokenizer(f\"{initial_prompt}{dummy_sentence} negative\")[\"input_ids\"])\n",
-    "print(opt_tokenizer(f\"{initial_prompt}{dummy_sentence} positive\")[\"input_ids\"])\n",
+    "print(llama_tokenizer(f\"{initial_prompt}{dummy_sentence} negative\")[\"input_ids\"])\n",
```

Comment:
Just curious: is there a reason why we needed to use the dummy sentence and initial prompt here instead of merely providing the label words as inputs to the tokenizer?

Reply:
That's a good question. Depending on the tokenizer, the ID of a word can be different if it's on its own vs. as part of a sentence. For example, having a space in front of it or not. Since we're using it at the end of a prompt, we consider the token ID as part of a sentence to make sure we get the right one.

Reply:
I'll put a comment to that affect in the notebook for clarity.

Reply:
Ah yes that totally makes sense!